lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
```

Comment:
The reason I found this "separating the big PR into small PR's" tedious is that some code is placed out of context and I have to re-explain every implementation reasoning that would otherwise make perfect sense as a whole, or redirect people to PR https://github.com/Shopify/semian/pull/54 :(